### PR TITLE
JAMES-1810 Register ReIndexer in JMX server

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/container/mailbox-adapter/src/main/java/org/apache/james/adapter/mailbox/ReIndexerManagement.java [hunk 1]
```diff
@@ -19,15 +19,19 @@
...
 import org.apache.james.mailbox.exception.MailboxException;
 import org.apache.james.mailbox.indexer.ReIndexer;
 import org.apache.james.mailbox.model.MailboxPath;
 
 public class ReIndexerManagement implements ReIndexerManagementMBean {
 
-    private final ReIndexer reIndexer;
+    private ReIndexer reIndexer;
 
-    public ReIndexerManagement(ReIndexer reIndexer) {
```
Comment (mbaechler) [inline server/container/mailbox-adapter/src/main/java/org/apache/james/adapter/mailbox/ReIndexerManagement.java]: why don't you keep the constructor injection ?

Comment (aduprat) [inline server/container/mailbox-adapter/src/main/java/org/apache/james/adapter/mailbox/ReIndexerManagement.java]: In order to respect what is done in other ManagerMBean (i'm sure this is working with spring that way)

Comment (aduprat) [inline server/container/mailbox-adapter/src/main/java/org/apache/james/adapter/mailbox/ReIndexerManagement.java]: And i tested it !

**Comment by mbaechler** [comment]
:+1:

**Pull request merged**